src/query.cpp

```diff
@@ -132,115 +132,90 @@ Rcpp::NumericVector transform_timestamp_points(
 }
 
 Rcpp::DataFrame transform_rows(qdb_point_result_t ** rows,
-    qdb_size_t rows_count,
-    qdb_size_t columns_count,
+    qdb_size_t row_count,
+    qdb_size_t column_count,
     const Rcpp::StringVector & columns)
 {
     assert(rows);
-    assert(rows_count > 0);
-    assert(columns_count > 0);
+    assert(column_count > 0);
 
-    Rcpp::List df; // {rows_count * columns_count};
+    Rcpp::DataFrame df; // {row_count * column_count};
 
-    for (qdb_size_t column_index = 0u; column_index < columns_count;
-         ++column_index)
+    if (row_count)
```

Comment:
```suggestion
    if (!row_count) return df;
```
?
You just need to set `df.attr("class") = "data.frame";` before.

Reply:
Done.